backend/services/auth.go

```diff
@@ -0,0 +1,59 @@
+package services
+
+import (
+	"backend/config"
+	"backend/models"
+	"backend/stores"
+	"time"
+
+	"github.com/golang-jwt/jwt/v4"
+	"golang.org/x/crypto/bcrypt"
+)
+
+type (
+	AuthService interface {
+		GenerateAccessToken(username string, password string, admin bool) (accessToken string, exp int64, err error)
+		GetUserById(id string) *models.User
+	}
+
+	authService struct {
+		*config.Config
+		*stores.Stores
+	}
+)
+
+func (s *authService) GenerateAccessToken(username string, password string, admin bool) (accessToken string, exp int64, err error) {
```

Comment:
[want]
```suggestion
func (s *authService) GenerateAccessToken(username, password string, admin bool) (string, int64, error) {
```

named return type は個人的には変数を追いづらくためあまり使わないほうがいい派です。
(defer で return する時など一部どうしようもない場合があるようですが)

日本語記事ですが、おおよそこのあたりの意見に賛成です。
https://zenn.dev/yuyu_hf/articles/c7ab8e435509d2#code-review-comments

代わりに doc コメントで帰り値に関する説明を提供したり Defined type で用途を制限した値を返すなどが良いかと思います。

あとついでですが、引数で同じ型のものが続く場合 (今回は string) は、冗長な記述を省略することができます。

Reply:
ありがとうございます！修正しました!